src/Domain/Donation.php

```diff
@@ -847,10 +840,31 @@ private function getConfirmedPledgeWithdrawalTotal(): string
             return '0.0';
         }
 
+        return $this->getWithdrawalTotalByFundType(Pledge::class);
+    }
+
+    /**
+     * @return numeric-string
+     */
+    private function getPreAuthorizedChampionWithdrawalTotal(): string
```

Comment:
I might have to step back to take a wider look, but this seems confusing as a concept, and also hard to see when just focusing on this class why it's only ChampionFund withdrawals that we want a preauthed figure for and not the other two types of Fund.

Reply:
OK I see that returning zero depending on the status is following an existing pattern even if it might be surprising.

This and the similar functions are all always called together, so I think it would be worth combining them into one that just iterates through `$this->fundingWithdrawals` once and returns all the totals together. This would be a bit more performant but the main advantage is that then it can throw (and possibly also fail a static check) if it encounters any fund of an unexpected type. 

Currently we'd ignore any fund of an unexpected type, which I think may be a bug in the case of a topup fund.